istioctl/cmd/istioctl/inject.go

```diff
@@ -252,21 +253,22 @@ istioctl kube-inject -f deployment.yaml -o deployment-injected.yaml --injectConf
 			}
 
 			var sidecarTemplate string
-			if injectConfigFile != "" {
-				injectionConfig, err := ioutil.ReadFile(injectConfigFile) // nolint: vetshadow
-				if err != nil {
-					return err
-				}
-				var config inject.Config
-				if err := yaml.Unmarshal(injectionConfig, &config); err != nil {
-					return err
+
+			// hub and tag params only work with ISTIOCTL_USE_BUILTIN_DEFAULTS
+			// so must be specified together. hub and tag no longer have defaults.
+			if hub != "" || tag != "" {
```

Comment:
Why adding this check here? Does only checking `ISTIOCTL_USE_BUILTIN_DEFAULTS` enough?